pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedConsumerImpl.java

```diff
@@ -355,6 +385,8 @@ void messageReceived(Message message) {
             if (log.isDebugEnabled()) {
                 log.debug("[{}][{}] Received message from partitioned-consumer {}", topic, subscription, message.getMessageId());
             }
+
+            unAckedMessageTracker.add((MessageIdImpl) message.getMessageId());
```

Comment:
Im not sure about this, I think that tracker should add messages only if pendingReceives.isEmpty(). It is ok to track messages on a consumer which is not on receive call?. Also will be more efficient.